endpoint-root/endpoint-core/src/main/java/org/divy/common/bo/endpoint/hypermedia/AbstractHyperMediaEndpoint.java

```diff
@@ -3,43 +3,48 @@
 import org.divy.common.bo.BusinessObject;
 import org.divy.common.bo.business.BOManager;
 import org.divy.common.bo.endpoint.AbstractCRUDEndpoint;
-import org.divy.common.bo.endpoint.hypermedia.association.AbstractAssociations;
+import org.divy.common.bo.endpoint.hypermedia.association.AssociationsHandler;
+import org.divy.common.bo.endpoint.hypermedia.association.Association;
 import org.divy.common.bo.query.Query;
 import org.divy.common.bo.rest.HyperMediaMapper;
 import org.divy.common.bo.rest.response.ResponseEntityBuilderFactory;
 import org.divy.common.exception.NotFoundException;
 
 import javax.validation.constraints.NotNull;
 import java.io.Serializable;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Optional;
+import java.util.*;
 
 public abstract class AbstractHyperMediaEndpoint<B extends BusinessObject<I>
         , E extends Representation
         , I extends Serializable
         , R>
         extends AbstractCRUDEndpoint<E, I, R> {
 
-    AbstractHyperMediaEndpoint(ResponseEntityBuilderFactory<E, R> responseEntityBuilderFactory) {
+    private final AssociationsHandler<B,I> associationsHandler;
+
+    public AbstractHyperMediaEndpoint(ResponseEntityBuilderFactory<E, R> responseEntityBuilderFactory, AssociationsHandler<B,I> associationsHandler) {
         super(responseEntityBuilderFactory);
+        this.associationsHandler = associationsHandler;
     }
 
     public abstract BOManager<B, I> getManager();
 
-    R updateRelation(I id, String relation) {
-        //TODO Implement
+    protected R updateRelation(I id, String relation) {
         return responseEntityBuilderFactory.update().build();
     }
 
     public R readRelation(I id, String relation) {
 
         B entity = getManager().get(id);
 
-        Optional<Object> optionalEntityRelation = this.getAssociations().getAssociation(relation).read(entity);
+        final Association<B, I> biAssociation = this.getAssociationsHandler().getAssociation(relation)
+                .orElseThrow(() -> new NotFoundException("Association " + relation + " not found"));
 
-        if(optionalEntityRelation.isPresent()) {
-            return responseEntityBuilderFactory.read((E)optionalEntityRelation.get()).build();
+        Optional<Object> optionalEntityRelation = biAssociation.read(entity);
+
+        if (optionalEntityRelation.isPresent()) {
+            final R response = responseEntityBuilderFactory.list((Collection<E>) optionalEntityRelation.get()).build();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "response". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)